tests/test_xsrftool.py

```diff
@@ -0,0 +1,50 @@
+import datetime
+import unittest
+
+import config
+import utils
+
+
+class XsrfToolTests(unittest.TestCase):
+
+    TEST_NOW = datetime.datetime(2010, 1, 31, 18, 0, 0)
+
+    def setUp(self):
+        utils.set_utcnow_for_test(XsrfToolTests.TEST_NOW)
+
+    def testGenerateAndVerifyGoodToken(self):
+        config.set(xsrf_token_key='abcdef')
+        tool = utils.XsrfTool()
+        token = tool.generate_token(12345, 'test_action')
+        self.assertTrue(tool.verify_token(token, 12345, 'test_action'))
```

Comment:
just in case it's relevant:
self.assertIs(True, <expr>) actually tests that <expr> is True
whereas
self.assertTrue(<extr>) tests that bool(<expr>) is True

so if you want to detect functions suddenly returning [] instead of False, you can use assertIs(True/False)
https://docs.python.org/2/library/unittest.html#unittest.TestCase.assertTrue

Reply:
Thanks, that's good to know, but I don't think it's a concern here: the XSRF tool's job is just to give a thumbs-up/thumbs-down (even if it did start returning an empty list or something, its callers would still interpret it the same way the test does).